### PR TITLE
Added a global option to use BGzip codec to read .gz files

### PR DESCRIPTION
**Pull request by lfrancioli**

### CODE REVIEW

File: src/main/scala/org/broadinstitute/hail/driver/Main.scala [hunk 1]
```diff
@@ -35,8 +35,12 @@ object SparkManager {
 
       conf.set("spark.serializer", "org.apache.spark.serializer.KryoSerializer")
       _sc = new SparkContext(conf)
-      _sc.hadoopConfiguration.set("io.compression.codecs",
-        "org.apache.hadoop.io.compress.DefaultCodec,org.broadinstitute.hail.io.compress.BGzipCodec,org.apache.hadoop.io.compress.GzipCodec")
+      if(gzAsBgz)
+        _sc.hadoopConfiguration.set("io.compression.codecs",
+          "org.apache.hadoop.io.compress.DefaultCodec,org.broadinstitute.hail.io.compress.BGzipCodec, org.broadinstitute.hail.io.compress.BGzipCodecGZ")
```
Comment (tpoterba) [inline src/main/scala/org/broadinstitute/hail/driver/Main.scala]: How does this handle gzipped files that are actually gzipped?  Suppose that you import a gz (bgz) vcf and then annotate with a small tsv.gz (actually gz)

Comment (lfrancioli) [inline src/main/scala/org/broadinstitute/hail/driver/Main.scala]: It doesn't: it's literally "gz as bgz".

Comment (lfrancioli) [inline src/main/scala/org/broadinstitute/hail/driver/Main.scala]: To be more precise, it throws a java.util.zip.ZipException: null

Comment (tpoterba) [inline src/main/scala/org/broadinstitute/hail/driver/Main.scala]: I think we really want it on ImportVCF, right?  I think you can make that happen by copying the spark context and setting the io.compression.codecs on the copy, then using that

**Comment by lfrancioli** [comment]
Yes, I  like that!

**Pull request closed without merging**